### PR TITLE
CC32xx firmware update demo

### PR DESCRIPTION
**Pull request by ashvayka**

### CODE REVIEW

File: fwupdate/source/cc32xx/README [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+This demo application illustrates configuration feature using the Kaa C SDK.
+To read more about the logging feature visit https://docs.kaaproject.org/display/KAA/Logging.
+To read more about the configuration feature visit https://docs.kaaproject.org/display/KAA/Configuration.
+To read more about the profiling feature visit https://docs.kaaproject.org/display/KAA/Endpoint+profiling.
+
+************************************
+INSTALLATION
+************************************
+
+*TODO*
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/README]: Please replace this TODO with all required instructions

File: fwupdate/source/cc32xx/build.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+PROJECT_HOME=$(pwd)
+BUILD_DIR="build"
+LIBS_PATH="libs"
+KAA_LIB_PATH="$LIBS_PATH/kaa"
+KAA_C_LIB_HEADER_PATH="$KAA_LIB_PATH/src"
+KAA_CPP_LIB_HEADER_PATH="$KAA_LIB_PATH/kaa"
+KAA_SDK_TAR="kaa-client*.tar.gz"
+
+#Wifi settings
+SSID="xxx"
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/build.sh]: Maybe leave this empty and check in build script that this settings are specified?

File: fwupdate/source/cc32xx/platform/cc32xx/app.ld [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/*****************************************************************************
+* blinky.ld
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/platform/cc32xx/app.ld]: What is this file for? why ids do not match?

File: fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c [hunk 4]
```diff
@@ -0,0 +1,758 @@
...
+ * limitations under the License.
+ */
+
+#include <time.h>
+#include "cc32xx_support.h"
+
+#define DEMO_UNUSED(x) if(x){}
+
+#define HTTP_BUF_LEN 1440
+#define FIRMWARE_FILE_PATH "/sys/mcuimg1.bin"
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c]: If this filename does not match to default, please document this in installation instructions.

File: fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c [hunk 5]
```diff
@@ -0,0 +1,758 @@
...
+
+unsigned long  g_ulStatus = 0;//SimpleLink Status
+unsigned long  g_ulGatewayIP = 0; //Network Gateway IP address
+unsigned char  g_ucConnectionStatus = 0;
+unsigned char  g_ucSimplelinkstarted = 0;
+unsigned long  g_ulIpAddr = 0;
+
+const char g_acSNTPserver[][30] =
+{
+    "dmz0.la-archdiocese.net"
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c]: what is this?

File: fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c [hunk 6]
```diff
@@ -0,0 +1,758 @@
...
+};
+
+static unsigned crc_32_tab[] = {
+    0x00000000, 0x77073096, 0xee0e612c, 0x990951ba, 0x076dc419, 0x706af48f,
+    0xe963a535, 0x9e6495a3,	0x0edb8832, 0x79dcb8a4, 0xe0d5e91e, 0x97d2d988,
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c]: redundant spaces?

File: fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c [hunk 7]
```diff
@@ -0,0 +1,758 @@
+/*
```
Comment (ashvayka) [inline fwupdate/source/cc32xx/platform/cc32xx/cc32xx_support.c]: Why do we need this file in example?

File: fwupdate/resources/profile_schema.avsc [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                    "namespace":"org.kaaproject.demo.fwupdate",
+                    "fields":[  
+                        { "name":"type", "type": {  
+                                "type":"enum",
+                                "name":"SensorType",
+                                "namespace":"org.kaaproject.demo.fwupdate",
+                                "symbols":[ "TEMPERATURE", "HUMIDITY", "LIGHT", "PRESSURE", "ACOUSTIC"]
+                            }
+                        },
+                        { "name":"model", "type": { "type":"string", "avro.java.string":"String"}}
```
Comment (dyosick) [inline fwupdate/resources/profile_schema.avsc]: I propose to use some JSON formatter, e.g. https://jsonformatter.curiousconcept.com/

File: fwupdate/resources/profile_schema.avsc [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+                    ]
+                }
+            }
+        },
+        { "name":"firmwareVersion", "type": {
```
Comment (ghost) [inline fwupdate/resources/profile_schema.avsc]: I propose to use some JSON formatter, e.g. https://jsonformatter.curiousconcept.com/

File: fwupdate/source/cc32xx/build.sh [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    cd $PROJECT_HOME &&
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR" &&
+    cp "$KAA_LIB_PATH/$BUILD_DIR/"libkaa* "$PROJECT_HOME/$BUILD_DIR/" &&
+    cd $BUILD_DIR
+    ENV_VAR=" -DKAA_PLATFORM=cc32xx -DAPP_NAME=$APP_NAME"
+	if [ "$(expr substr $(uname -s) 1 9)" == "CYGWIN_NT" ]; then
+		ENV_VAR=$ENV_VAR" -DKAA_TOOLCHAIN_PATH=c:/cygwin64/opt/kaa"
+	fi
+	
+	#source PATH=$PATH:/opt/kaa/gcc-arm-none-eabi/bin
```
Comment (ghost) [inline fwupdate/source/cc32xx/build.sh]: Do we need these commented lines?

**Pull request merged**